packages/backend/src/models/ProjectModel/ProjectModel.ts

```diff
@@ -1208,6 +1211,27 @@ export class ProjectModel {
                 newId: newDashboardVersions[i].dashboard_version_id,
             }));
 
+            const dashboardViews = await trx(DashboardViewsTableName).whereIn(
+                'dashboard_version_id',
+                dashboardVersionIds,
+            );
+
+            Logger.debug(
+                `Duplicating ${dashboardViews.length} dashboard views on ${previewProjectUuid}`,
+            );
+
+            if (dashboardViews.length > 0) {
+                await trx(DashboardViewsTableName).insert(
+                    dashboardViews.map((d) => ({
+                        ...d,
+                        dashboard_view_uuid: undefined,
+                        dashboard_version_id: dashboardVersionsMapping.find(
+                            (m) => m.id === d.dashboard_version_id,
+                        )?.newId!,
```

Comment:
so by passing this `newId` it gets the filters applied? 🤔

Reply:
When we duplicate an asset we keep a map of the old and new ids. `dashboardVersionsMapping` 

When we copy the filters row ( table `dashboard_views` ) we need to replace the old version id ( which is from the original dashboard version ) with the new one ( the newly created dashboard version ).

Database structure.
![Uploading Screenshot 2023-09-22 at 12.20.37.png…]()